sqlglot/optimizer/isolate_table_selects.py

```diff
@@ -12,7 +12,12 @@ def isolate_table_selects(expression, schema=None):
             continue
 
         for _, source in scope.selected_sources.values():
-            if not isinstance(source, exp.Table) or not schema.column_names(source):
+            if (
+                not isinstance(source, exp.Table)
+                or not schema.column_names(source)
+                or isinstance(source.parent, exp.Subquery)
+                or isinstance(source.parent.parent, exp.Table)
```

Comment:
The first added instance check ensures we don't expand `t` in `(t)` into a star query, whereas the second one ensures we similarly don't expand `t2` in `(t1 join t2)`.

Reply:
but isn't t2 a table? why would it be triggered? same with the second case, the second case 

t1 join t2, it's a table right?